quilt3distribute/dataset.py

```diff
@@ -252,30 +307,99 @@ def distribute(
 
                     # Update values to the logical key as they are set
                     for i, val in enumerate(v_ds.data[col].values):
+                        # Fully resolve the path
                         pk = Path(val).expanduser().resolve()
+
+                        # Just using val.name could result in files that shouldn't be grouped being grouped
+                        # Example column:
+                        # SourceReadpath
+                        # a/0.tiff
+                        # a/1.tiff
+                        # b/0.tiff
+                        # b/1.tiff
+                        # Even though there are four files, this would result in both a/0.tiff and b/0.tiff, and,
+                        # a/1.tiff and b/1.tiff being grouped together. To solve this we can prepend a the first couple
+                        # of characters from a hash of the fully resolved path to the logical key.
+                        prepend = hashlib.sha256(str(pk).encode("utf-8")).hexdigest()[:8]
```

Comment:
I think this should be a separate static method so that if you modify how this uniqueness is done it's much clearer